### DISCUSSION TITLE
getSession() in  getServerSideProps  with HTTPS not work!

### DISCUSSION THREAD

**Discussion by sky9413** [nextauthjs/next-auth#2604] [Help]
### Question 💬

Hello,

I cant use getSession()  in  getServerSideProps  with HTTPS.

is it normal? I try many times.

I will get it if use  HTTPS .  I cant getSession()  in  getServerSideProps 
```
__Secure-next-auth.callback-url
__Secure-next-auth.session-token
__Host-next-auth.csrf-toke
```

if use HTTP and I can getSession()  in  getServerSideProps is Okay
```
next-auth.callback-url
next-auth.session-token
next-auth.csrf-token
```

how can I fixed it on HTTPS getSession()  in  getServerSideProps?

### How to reproduce ☕️

I run the same code on http or https for test

if run with http, I can get the props.session
if run with **https**, I cannot get the props.session

```
import { getSession } from 'next-auth/client';

export default function Home(props) {
  console.log(props.session);
  return (
    <div>
      <h1>Server Side Rendering</h1>
    </div>
  );
}
export async function getServerSideProps(context) {
  return {
    props: {
      session: await getSession(context),
    },
  };
}

```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by doorman02** [comment]
Hi @sky9413 did you find a solution to this problem? I am having the same problem getSession returns undefined when using https

**Comment by JoaoOxOc** [comment]
@doorman02 @sky9413 
Hi, took me a lot of time to understand this issue. The solution could be one of the following:

- if the next-auth server cannot access the canonical URL defined in the env var NEXTAUTH_URL (like https://example.com is not reachable from the nextjs server host), another env var must be defined as localhost (mapping to the nextJS server host or docker container): NEXTAUTH_URL_INTERNAL=http://localhost:<your nextJS host port>. Check: https://next-auth.js.org/v3/configuration/options#nextauth_url_internal. This var will set next-auth server side actions to execute localhost (it makes sense, since a getServerSideProps() from nextJS calling getSession() for example will invoke a localhost action, doesn't need to route outside)

- If setting the internal var doesn't solve, could be due to secure cookies (HTTPS). In authoptions set `useSecureCookies: false,`: this will tell next-auth to never set the cookies as secure site only (HTTPS)
